lib/Conversion/TorchToLinalg/Uncategorized.cpp

```diff
@@ -770,19 +804,22 @@ static Value createLinalgPayloadCalculationForElementwiseOp(
       reciprocal.emitError("unimplemented: non-floating point dtype");
       return nullptr;
     }
-
-    Type elementType = payloadArgs[0].getType();
```

Comment:
the code above requiring input to be float can be deleted. The e2e test added for `reciprocal` should have failed because of this code. Could you double check if it's the case?

Reply:
I think the check above is for the return type, which will be a float because of the `RefineTypes` code, so the test should be passing

Reply:
Actually, the e2e test for `reciprocal` works for `int32` input. Because op's `getType()` returns the type of the first result of the Operation:
```C++
    ResultType getType() {
      auto resultTy = this->getOperation()->getResult(0).getType();
      return resultTy.template cast<ResultType>();
    }
```
And the test passed locally:
```
PASS - "ElementwiseReciprocalIntModule_basic"
```

Since we set the resulting dtype to float anyway, so I think the code should work for any input dtype which is not `null`. But I do worry about the case of 'null' input dtype.

Reply:
ha, you are right! The code just need to be deleted then.